libdnf/repo/repo.cpp

```diff
@@ -181,12 +181,15 @@ void Repo::Impl::load_available_repo(LoadFlags flags) {
     }
 
     if (any(flags & LoadFlags::COMPS)) {
-        auto path = get_metadata_path(RepoDownloader::MD_FILENAME_GROUP_GZ);
-        if (path.empty()) {
-            path = get_metadata_path(RepoDownloader::MD_FILENAME_GROUP);
+        auto md_filename = get_metadata_path(RepoDownloader::MD_FILENAME_GROUP_GZ);
+        if (md_filename.empty()) {
+            md_filename = get_metadata_path(RepoDownloader::MD_FILENAME_GROUP);
         }
-        if (!path.empty()) {
-            base->get_comps()->load_from_file(owner->get_weak_ptr(), path);
+
+        if (!md_filename.empty()) {
+            solv_repo.load_repo_ext(md_filename, RepodataType::COMPS);
+        } else {
+            logger.debug(fmt::format("no comps available for {}", config.get_id()));
```

Comment:
Do we need to report it? Comps are not mandatory part and they are often not described in repomd. Comps are only used in Fedora and RHEL repositories. If we will report it, shall we also report missing module.yaml?

Reply:
Well it's at debug level (which will not be on by default). I added this one just for consistency, this code needs more refactoring and I'm planning on going over the logging too. A message of some kind is a good idea when loading each of the extra data files, either when it's loading it or when it's not loading it :slightly_smiling_face: (or both if we'd want to be rigorous at the expense of being verbose).

Reply:
Then we have to logger any missing data type which are allowed to load. But I do afraid that it will represent a long list including module filer, file lists, updateinfo, additional data requested bu user and so on. I thing that the information what kind of optional data are not loaded has a low informative value.

Reply:
I am sorry I see your point. All other types of metadata have the same logging mechanism. What about to drop logging when some types of data are not available? We can discuss it later on.